### PR TITLE
CONN-1513: Added code to enable Agent Settings cache

### PR DESCRIPTION
**Pull request by nareshcgi**
1. Added logic to enable Agent Settings Cache
2. Added logic to configure Agent Settings Cache (two new properties in gateway.properties)
   a) AgentSettingsCacheRefreshTime --> Cache refresh time in milli seconds
   b)AgentSettingsCacheRefreshActive  --> to disable Agent settings cache

### CODE REVIEW

**Comment by alameluchidambaram** [comment]
Reviewed PR and looks good

File: Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/messagemonitoring/util/MessageMonitoringUtil.java [hunk 1]
```diff
@@ -600,4 +602,64 @@ private static int getPropertyIntegerValue(String propertyName) {
...
+     * properties file. If not found, returns false
+     *
+     * @return
+     */
+    public static boolean getAgentSettingsCacheRefreshActive(){
+        try {
+            String agentSettingsCacheRefreshActive = PropertyAccessor.getInstance().getProperty(AGENT_SETTINGS_CACHE_REFRESH_ACTIVE);
+            return (agentSettingsCacheRefreshActive != null) && (agentSettingsCacheRefreshActive.equals("true"));
+        } catch (PropertyAccessException ex) {
+            LOG.info("Proeprty Not found :" + AGENT_SETTINGS_CACHE_REFRESH_ACTIVE);
```
Comment (jsmith2) [inline Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/messagemonitoring/util/MessageMonitoringUtil.java:658]: Property misspelled in both logging statements.

File: Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/DirectAdapter.java [hunk 2]
```diff
@@ -133,23 +139,52 @@ private MessageProcessResult processAsDirectMessage(MimeMessage mimeMessage) {
...
+                //if the agent settings manager is running then stop it and start it again
+                if (gatewayState.isAgentSettingManagerRunning()) {
+                    LOG.trace("Stop Agent Settings Manager");
+                    gatewayState.stopAgentSettingsManager();
+                }
+                //start the Agent Settings Manager
+                LOG.trace("Start Agent Settings Manager");
+                gatewayState.startAgentSettingsManager();
+            } else {
+                LOG.trace("Agent Settings Manager not enabled");
```
Comment (jsmith2) [inline Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/DirectAdapter.java]: Think this log statement should be debug or info not trace.

File: Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/DirectAdapter.java [hunk 3]
```diff
@@ -133,23 +139,52 @@ private MessageProcessResult processAsDirectMessage(MimeMessage mimeMessage) {
...
      */
     protected SmtpAgent getSmtpAgent(URL url) {
+        
         if ((smtpAgent != null) && smtpAgent instanceof SmtpAgent) {
             return smtpAgent;
         } else {
             smtpAgent = SmtpAgentFactory.createAgent(url);
+            if (MessageMonitoringUtil.getAgentSettingsCacheRefreshActive()) {
+                GatewayState gatewayState = GatewayState.getInstance();
+                //Update timout seconds
```
Comment (jsmith2) [inline Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/DirectAdapter.java]: Timeout misspelled in comment.

**Pull request merged**